`hack/create-cluster-lima.sh` doesn't work on macOS (likely to be a bug of Lima)

```console
$ git clone https://github.com/rootless-containers/usernetes
$ cd usernetes
$ git checkout gen2-v20241203.0
$ ./hack/create-cluster-lima.sh
[...]
NFO[0000] Attempting to download the image              arch=x86_64 digest="sha256:05bbfe57d7701c685d8c65f4d3
4cebe947bc89e3509c4d8a2b9c77f39e91f3ca" location="https://cloud-images.ubuntu.com/releases/24.10/release-20241
109/ubuntu-24.10-server-cloudimg-amd64.img"
[...]
+ dockerd-rootless-setuptool.sh install
[ERROR] Missing system requirements. Run the following commands to
[ERROR] install the requirements and run this tool again.

########## BEGIN ##########
sudo sh -eux <<EOF
# Add subuid entry for suda
echo "suda:100000:65536" >> /etc/subuid
# Add subgid entry for suda
echo "suda:100000:65536" >> /etc/subgid
EOF
########## END ##########

+ journalctl --user --since '10 min ago'
Hint: You are currently not seeing messages from the system.
      Users in groups 'adm', 'systemd-journal' can see all messages.
      Pass -q to turn off this notice.
No journal files were opened due to insufficient permissions.
```

Lima: v1.0.2
Usernetes: gen2-v20241203.0
macOS: 15.1.1 (Intel)

--- a/hack/create-cluster-lima.sh
+++ b/hack/create-cluster-lima.sh
@@ -2,16 +2,23 @@
 set -eux -o pipefail
 
 : "${LIMACTL:=limactl --tty=false}"
+: "${LIMACTL_CREATE_ARGS:=}"
 : "${LIMA_TEMPLATE:=template://default}"
 : "${CONTAINER_ENGINE:=docker}"
 : "${LOCKDOWN_SUDO:=1}"
 
 guest_home="/home/${USER}.linux"
 
+if [ "$(id -u)" -le 1000 ]; then
+	# In --plain mode, UID has to be >= 1000 to populate subuids
+	# https://github.com/lima-vm/lima/issues/3001
+	LIMACTL_CREATE_ARGS="${LIMACTL_CREATE_ARGS} --set=.user.uid=1001"
+fi
+
 # Create Rootless Docker hosts
 for host in host0 host1; do
 	# Set --plain to minimize Limaism
-	${LIMACTL} start --plain --network lima:user-v2 --name="${host}" "${LIMA_TEMPLATE}"
+	${LIMACTL} start --plain --network lima:user-v2 --name="${host}" ${LIMACTL_CREATE_ARGS} "${LIMA_TEMPLATE}"
 	${LIMACTL} copy -r "$(pwd)" "${host}:${guest_home}/usernetes"
 	${LIMACTL} shell "${host}" sudo CONTAINER_ENGINE="${CONTAINER_ENGINE}" "${guest_home}/usernetes/init-host/init-host.root.sh"
 	# Terminate the current session so that the cgroup delegation takes an effect. This command exits with status 255 as SSH terminates.

